src/lazy/encoder/binary/v1_0/writer.rs

```diff
@@ -156,6 +158,14 @@ impl<W: Write> LazyRawWriter<W> for LazyRawBinaryWriter_1_0<W> {
         self.output.write_all(&[0xE0, 0x01, 0x00, 0xEA])?;
         Ok(())
     }
+
+    fn macro_table(&self) -> &WriterMacroTable {
+        &EMPTY_MACRO_TABLE
+    }
+
+    fn macro_table_mut(&mut self) -> Option<&mut WriterMacroTable> {
+        None
+    }
```

Comment:
🪧 `LazyRawWriter` implementations are now required to own their own `WriterMacroTable`. 

As the trait method doc comments explain (later in the diff), for read-only access there is no `Option` type because that's a very hot path. Ion 1.0 writers can return a reference to the global static `EMPTY_MACRO_TABLE` to satisfy the API.

For mutable access (only required to modify the encoding context, which is comparatively rare), implementations can return an `Option<&mut WriterMacroTable>`. This forces the managed writer to confirm on each access that there's a macro table to modify.